src/backend/routers/auth.py

```diff
@@ -317,3 +317,74 @@ def log_and_redirect_err(error_message: str):
     response = RedirectResponse(redirect_uri)
 
     return response
+
+
+@router.delete("/tool/auth/{tool_id}")
+async def delete_tool_auth(
+    tool_id: str,
+    request: Request,
+    session: DBSessionDep,
+    ctx: Context = Depends(get_context),
+):
+    """
+    Endpoint to delete Tool Authentication.
+
+    If completed, the corresponding ToolAuth for the requesting user is removed from the DB.
+
+    Args:
+        tool_id (str): ToolID to be deleted for the user. (eg. google_drive) Should be one of the values listed in the ToolName string enum class.
+        request (Request): current Request object.
+        session (DBSessionDep): Database session.
+        ctx (Context): Context object.
+
+    Returns:
+        {"status": "success"} on successful deletion.
+
+    Raises:
+        HTTPException: If there was an error deleting the tool auth.
+    """
+
+    logger = ctx.get_logger()
+
+    def log_and_return_error(error_message: str):
```

Comment:
2 things:

1. Let's rename this to `log_and_raise_exception` to better reflect what it performs
2. Let's move this out of the routers, and into the `services` folder, perhaps in the logger subfolder?